samza-core/src/test/java/org/apache/samza/processor/TestStreamProcessor.java

```diff
@@ -43,53 +48,85 @@
 import static org.mockito.Mockito.when;
 
 public class TestStreamProcessor {
+  private Map<String, Boolean> processorListenerState;
+  private static final String ON_START = "onStart";
+  private static final String ON_SHUTDOWN = "onShutdown";
+  private static final String ON_FAILURE = "onFailure";
+
+  @Before
+  public void before() {
+    processorListenerState = new HashMap<String, Boolean>() {
+      {
+        put(ON_START, false);
+        put(ON_FAILURE, false);
+        put(ON_SHUTDOWN, false);
+      }
+    };
+  }
+
+  @After
+  public void after() {
+    processorListenerState.clear();
+  }
 
   class TestableStreamProcessor extends StreamProcessor {
     private final CountDownLatch containerStop = new CountDownLatch(1);
     private final CountDownLatch runLoopStartForMain = new CountDownLatch(1);
-    private SamzaContainer containerReference = null;
+    public SamzaContainer container = null;
```

Comment:
Is it possible to make this private? I didn't see an obvious reason why it is public, but it's quite possible I missed it.